### DISCUSSION TITLE
New contributor - Project suggestion

### DISCUSSION THREAD

**Discussion by dynamic-queries** [CliMA/Oceananigans.jl#3519] [Computational science!]
Hello CliMA community
I have a background in applied math and CSE and would like to contribute  to the project.
Can you suggest a nice starter project that is more numerics inclined ?

**Comment by glwagner** [comment]
Hi @dynamic-queries ! It's best to pair a development project with an application that you are interested in, which will help drive and motivate the changes. What is your background and interests?

Are you an experienced user of Oceananigans yet? Having some experience setting up more complex simulations (going beyond the simple examples provided) will provide some valuable experience both in the internals and also in understanding the philosophy of our user interface.

**Comment by dynamic-queries** [comment]
@glwagner Thanks for the speedy reply.

I am interested in coupling ice-sheet and ocean models, with the goal of understanding the melting rates of the former. Of particular interest is investigating (numerically) bifurcations of these coupled PDEs, in the sense of https://www.pnas.org/doi/10.1073/pnas.2024192118 , where a simple toy model is considered.

As I mentioned I have a background in computational science. So, I am also interested in finding good bases functions to discretize the space domain of the PDEs; in the sense of https://www.sciencedirect.com/science/article/abs/pii/S0021999117304916 

I have worked through the tutorials. Although, I must admit that I have not ventured into setting up my own models.

I hope this helps.

**Reply by glwagner** [reply]
A coupled ice sheet / ocean modeling project might be possible with Oceananigans, though it would require some careful thought to find a project that is both feasible (ie not too complicated) and leverages Oceananigans strengths (the cited reference uses a toy model for which Oceananigans wouldn't be useful).

On PDE discretization, we use finite volume methods, so there isn't much scope to experiment with new bases for spatial discretization.

This is a long shot, but you may be interested in working with @fberonvera to develop a solver for the quasi-geostrophic (and "thermal quasi-geostrophic") equations. I'm not sure this work would constitute a direct contributions to Oceananigans (it would probably make sense to develop this solver in another repo, even if it uses a lot of Oceananigans objects / abstractions). There's a little bit of discussion on https://github.com/CliMA/Oceananigans.jl/discussions/3478 about what this might entail. The main piece of work that's needed is to develop a new Poisson solver for the quasi-geostrophic streamfunction. We have Poisson solvers for pressure, but the finite volume discretization of pressure is different than what's needed for the QG streamfunction, and we also require different boundary conditions.

Like anything Oceananigans numerics, this delves into the esotera of C-grid finite volume discretizations...

**Comment by glwagner** [comment]
@simone-silvestri might have additional thoughts.

**Comment by dynamic-queries** [comment]
> A coupled ice sheet / ocean modeling project might be possible with Oceananigans, though it would require some careful thought to find a project that is both feasible (ie not too complicated) and leverages Oceananigans strengths (the cited reference uses a toy model for which Oceananigans wouldn't be useful).

I agree. I will mull over this for the next week; and get back to you then.
